File: app/src/main/java/com/khmelenko/lab/travisclient/adapter/BranchesListAdapter.java
package com.khmelenko.lab.travisclient.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.khmelenko.lab.travisclient.R;
import com.khmelenko.lab.travisclient.adapter.viewholder.BuildViewHolder;
import com.khmelenko.lab.travisclient.network.response.Branch;
import com.khmelenko.lab.travisclient.network.response.Branches;
import com.khmelenko.lab.travisclient.network.response.Commit;

/**
 * List adapter for branches
 *
 * @author Dmytro Khmelenko
 */
public final class BranchesListAdapter extends RecyclerView.Adapter<BuildViewHolder> {

    private Branches mBranches;
    private final OnListItemListener mListener;

    public BranchesListAdapter(Branches branches, OnListItemListener listener) {
        mBranches = branches;
        mListener = listener;
    }

    @Override
    public BuildViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_build_view, parent, false);
        return new BuildViewHolder(v, mListener);
    }

    @Override
    public void onBindViewHolder(BuildViewHolder holder, int position) {
        if (mBranches != null) {
            Branch branch = mBranches.getBranches().get(position);
            Commit relatedCommit = null;
            for (Commit commit : mBranches.getCommits()) {
                if (branch.getCommitId() == commit.getId()) {
                    relatedCommit = commit;
                    break;
                }
            }

            holder.mBuildView.setState(branch);
            holder.mBuildView.setCommit(relatedCommit);
        }
    }

    @Override
    public int getItemCount() {
        return mBranches != null ? mBranches.getBranches().size() : 0;
    }

    public void setBranches(Branches repoStatus) {
        mBranches = repoStatus;
    }

}

File: app-v3/src/main/java/com/khmelenko/lab/varis/repodetails/BranchesListAdapter.kt
package com.khmelenko.lab.varis.repodetails

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.khmelenko.lab.varis.R
import com.khmelenko.lab.varis.adapter.viewholder.BuildViewHolder
import com.khmelenko.lab.varis.network.response.Branches

/**
 * List adapter for branches
 *
 * @author Dmytro Khmelenko
 */
class BranchesListAdapter(private var branches: Branches?,
                          private val listener: (Int) -> Unit
) : RecyclerView.Adapter<BuildViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BuildViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_build_view, parent, false)
        return BuildViewHolder(v, listener)
    }

    override fun onBindViewHolder(holder: BuildViewHolder, position: Int) {
        if (branches != null) {
            val branch = branches!!.branches[position]
            val relatedCommit = branches!!.commits.firstOrNull { it.id == branch.commitId }

            holder.buildView.setState(branch)
            holder.buildView.setCommit(relatedCommit)
        }
    }

    override fun getItemCount(): Int {
        return if (branches != null) branches!!.branches.size else 0
    }

    /**
     * Sets branches
     *
     * @param branches Branches
     */
    fun setBranches(branches: Branches?) {
        this.branches = branches
    }
}
